File: stress/t/arr_copy_elemwise.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t i,n,t,u,sadd,smul,scopy;
  int64_t A[30],B[30],C[30],D[30];
  n=30;
  for(i=0;i<n;i++){ A[i]=i*2+1; B[i]=i*i-5; }
  for(i=0;i<n;i++){ t=A[i]; C[i]=t; }
  for(i=0;i<n;i++){ t=A[i]; u=B[i]; D[i]=t+u; }
  scopy=0; sadd=0; smul=0;
  for(i=0;i<n;i++){
    t=C[i]; scopy=scopy+t;
    u=D[i]; sadd=sadd+u;
    t=A[i]; u=B[i]; smul=smul+t*u;
  }
  printf("copy="); printf("%ld",(long)scopy); printf("\n");
  printf("add="); printf("%ld",(long)sadd); printf("\n");
  printf("mul="); printf("%ld",(long)smul); printf("\n");
  return 0;
}

File: stress/t/arr_copy_elemwise.f
      PROGRAM ARRCOPYEW
      INTEGER*8 I,N,T,U,SADD,SMUL,SCOPY
      INTEGER*8 A(0:29),B(0:29),C(0:29),D(0:29)
      N=30
      I=0
   10 IF (I.GE.N) GOTO 20
      A(I)=I*2+1
      B(I)=I*I-5
      I=I+1
      GOTO 10
   20 CONTINUE
      I=0
   30 IF (I.GE.N) GOTO 40
      T=A(I)
      C(I)=T
      I=I+1
      GOTO 30
   40 CONTINUE
      I=0
   50 IF (I.GE.N) GOTO 60
      T=A(I)
      U=B(I)
      D(I)=T+U
      I=I+1
      GOTO 50
   60 CONTINUE
      SCOPY=0
      SADD=0
      SMUL=0
      I=0
   70 IF (I.GE.N) GOTO 80
      T=C(I)
      SCOPY=SCOPY+T
      U=D(I)
      SADD=SADD+U
      T=A(I)
      U=B(I)
      SMUL=SMUL+T*U
      I=I+1
      GOTO 70
   80 CONTINUE
      WRITE(*,'(A,I0)') 'copy=',SCOPY
      WRITE(*,'(A,I0)') 'add=',SADD
      WRITE(*,'(A,I0)') 'mul=',SMUL
      END
